charmcraft/commands/build.py

```diff
@@ -145,6 +146,9 @@ def __init__(self, args, config):
         self.entrypoint = args["entrypoint"]
         self.requirement_paths = args["requirement"]
         self.force_packing = args["force"]
+        self.debug = args["debug"]
+        self.shell = args["shell"]
+        self.shell_after = args["shell_after"]
```

Comment:
Let's talk about these three command line options for this functionality.

On one hand, is this the same `--debug` that we'll use to control "extreme verbosity" in the tool output, or we have names collision?

On the other hand, what about combinations between these options? I don't see any control of the user not passing `--shell` and `--debug` at the same time, for example, and one makes the other to not work, in the code. 

Also I don't like that `--shell-after` is explicitly after, while `--shell` does not indicate when it happens. Also, `debug` will open a shell, while the name does not indicate that at all...

What if we only add a `--shell=WHEN` option, with three possible values: `before`, `after`, and `exception`?

Reply:
The `--debug` clash is something we need to address. It will happen in all tools using craft-cli and providers+parts.